### ISSUE TITLE
how to share some basic settings and packages between two different images?

### ISSUE THREAD

**Issue by fedelibre** [systemd/mkosi#294]
mkosi version 4 installed from Fedora 29 repository.

## Use case

I'm building two Debian images: one is a simple container to be used on the command line, the other is a image containing a desktop environment to be run on QEMU or similar.
Both images share several packages, but the "desktop image" has a few more GUI packages, such as the browser.

Currently the packages in common are duplicated in both files. This means that I have *two files to maintain instead of one*. It's not a big deal, but I'd like to avoid duplication if possible.

## My attempts

Here's what I tried:

1. pass two mkosi files using --default option doesn't work. Only one file is allowed.
2. I've tried placing a mkosi.base file in `mkosi.default.d/`, but it's ignored when I build the image with `mkosi --default mkosi.desktop`.
3. pass the base packages on the `--package` command-line option, while specific packages for each image are in their default config file.

The third option works, even if I don't like it much as I have to type a long list of packages on the command line. It's not an issue when you use a Makefile or similar, but perhaps mkosi may handle it better? A couple of ideas:

- I haven't tried, but something like `--package=my-list.txt` might be already an improvement.
- another option could be some kind of `[include]` section in default config file

**Comment by fedelibre** [comment]
> I haven't tried, but something like `--package=my-list.txt` might be already an improvement.

Nope, if I save a comma separated list of packages in a file called `main-packages.txt`, the text file is not *expanded*:

```
Packages: main-packages.txt
```

**Comment by lucasdemarchi** [comment]
Hummn. I *think* this should work:

mkosi.minimal.d/
mkosi.desktop.d/

Then you add a 00-mkosi.base in one dir and a symlink to that file in the other. Any additional file can be added on the directories to differentiate the builds

**Comment by fedelibre** [comment]
Ok. 
Here's the tree:

```
[mkosi]$ tree mkosi.minimal.d/
mkosi.minimal.d/
├── 00-mkosi.base
└── 01-mkosi.container

0 directories, 2 files
[mkosi]$ 
[mkosi]$ 
[mkosi]$ tree mkosi.desktop.d/
mkosi.desktop.d/
├── 00-mkosi.base -> ../mkosi.minimal.d/00-mkosi.base
└── 01-mkosi.desktop

0 directories, 2 files
```

How can I call mkosi now to generate the two images?

**Comment by fedelibre** [comment]
Ok, it works!
Now I see the analogy with mkosi.default and mkosi.default.d. The README doesn't say that you can use any name which follows this scheme: so calling mkosi.NAME will pick up also the files in mkosi.NAME.d, allowing for multiple configuration files in the same directory.

The symlink is also a good hint to avoid file duplications.

I may send a PR to improve the README if you think it's worth.
